pkg/storage/stores/shipper/indexshipper/tsdb/single_file_index.go

```diff
@@ -365,11 +368,10 @@ func (i *TSDBIndex) Volume(
 		for k := range labelsToMatch {
 			by[k] = struct{}{}
 		}
-		if len(labelsToMatch) > 0 {
-			for k := range labelsToMatch {
-				by[k] = struct{}{}
-			}
+		for _, k := range targetLabels {
+			by[k] = struct{}{}
```

Comment:
this should be the same as `labelsToMatch`. see https://github.com/grafana/loki/blob/9eb557cb3757e80083d47ad493629775722a2d9b/pkg/util/series_volume.go#L34